pkg/resources/pciNetDevice.go

```diff
@@ -60,6 +61,15 @@ func NewPciNetDevice(pciDevice *ghw.PCIDevice, rFactory types.ResourceFactory) (
 		Health: pluginapi.Healthy,
 	}
 
+	linkType := ""
+	if len(ifName) > 0 {
+		la, err := utils.GetLinkAttrs(ifName)
+		if err != nil {
+			return nil, err
```

Comment:
I'd prefer not to return nil for not detecting linktype here. Maybe add linktype info only if we can get it otherwise leave it empty like we do for ifName. There are times when a VF may not be represented as a netdevice thus those info won't be available. But "selectors" will be able to capture only those devices that have these information when applied.

Reply:
For ifName, indeed you may not have a netdevice in case the device is not bounded to a network driver (hence ifName will be empty).
However, if we get into this condition then we have a netdevice (ifName is not empty)
Failing to get link attributes may point to some system fault, are we OK with ignoring that ?

Reply:
Good call. I think we can safely assume that if we can get a valid interface name then link attributes should be available. I think it's good idea to return the err. The discovery method logs this error anyway. Maybe we could add some more info into the error before returning?

Reply:
So, utils.GetLinkAttrs() returns an informative error in case of failure that includes the netlink error:

`func GetLinkAttrs(ifName string) (*nl.LinkAttrs, error) {
	link, err := getLinkByName(ifName)
	if err != nil {
		return nil, fmt.Errorf("error getting link attributes for net device %s %v", ifName, err)
	}
	return link.Attrs(), nil
}`

It will propagate up the stack and will be reflected in the discovery method.